### ISSUE TITLE
Update kube-watch.js scripts to support newer versions of Kubernetes

### ISSUE THREAD

**Issue by mixecan** [nanit/kubernetes-graphite-cluster#21]
Is there any plan to update kube-watch.js scripts in order for the project to be deployable on newer kubernetes?
The scripts currently use version 3.12 of `kubernetes-client`, whilst current version is 5.3.0, and they break on k8s v1.9

**Comment by erez-rabih** [comment]
Thanks for bringing this up
I'd be happy to get a PR for that

**Comment by erez-rabih** [comment]
Might be resolved by this commit: https://github.com/nanit/kubernetes-graphite-cluster/commit/7cdb6ea179b0122f86351f7a46ad6b37806d4cad

Would be nice if you can approve it

**Comment by mixecan** [comment]
Thank you erez. Have you tested it? I believe that the latest version of kubernetes-client expose a different API from version 3.12.
I can have a look in the next days :-)

**Comment by sholsapp** [comment]
I don't think this resolved the issue. I recently checked out at head and when running see this in the logs for one of the statsd nodes:

```
error: Forever detected script exited with code: 1
error: Script restart attempt #2
undefined
/app/statsd/kube-watch.js:26
  console.log(result.subsets.length)
                     ^

TypeError: Cannot read property 'subsets' of undefined
    at /app/statsd/kube-watch.js:26:22
    at /app/statsd/node_modules/kubernetes-client/lib/base.js:13:14
    at Request.request [as _callback] (/app/statsd/node_modules/kubernetes-client/lib/request.js:65:7)
    at Request.self.callback (/app/statsd/node_modules/request/request.js:185:22)
    at emitTwo (events.js:126:13)
    at Request.emit (events.js:214:7)
    at Request.<anonymous> (/app/statsd/node_modules/request/request.js:1157:10)
    at emitOne (events.js:116:13)
    at Request.emit (events.js:211:7)
    at IncomingMessage.<anonymous> (/app/statsd/node_modules/request/request.js:1079:12)
error: Forever detected script exited with code: 1
29 Jul 17:27:53 - WARNING: [20] Warning: No backend statsd nodes available!
29 Jul 17:28:05 - WARNING: [20] Warning: No backend statsd nodes available!
29 Jul 17:28:07 - WARNING: [20] Warning: No backend statsd nodes available!
```
FWIW, I had to make a few modifications to the graphite statefule set configurations, but that should be unrelated to this issue.

**Comment by sholsapp** [comment]
Ah, I missed a previous log message that was the root problem.

```
Received update: {                                                                                    
  "kind": "Status",                                                                  
  "apiVersion": "v1",                                                                                                                                         
  "metadata": {},                                                                                                                                             
  "status": "Failure",                                                                                                                                        
  "message": "endpoints is forbidden: User \"system:serviceaccount:default:default\" cannot watch endpoints at the cluster scope",                            
  "reason": "Forbidden",                                                                                                                                      
  "details": {                                                                          
    "kind": "endpoints"                                                                                                                                      
  },                                                                                   
  "code": 403                                                                                                                                                
}    
```

Still, it seems the kube-watch.js script has some buggy code in it, but after addressing above issue, the cluster seems to be working/plotting metrics.

**Comment by erez-rabih** [comment]
@sholsapp can you please submit a PR with your changes so that other people can enjoy the fix as well?

**Comment by erez-rabih** [comment]
closed by #23
